nbgrader/tests/preprocessors/test_clearmarkscheme.py

```diff
@@ -108,3 +108,65 @@ def test_preprocess_notebook(self, preprocessor):
         """Is the test notebook processed without error?"""
         nb = self._read_nb(os.path.join("files", "test_taskcell.ipynb"))
         preprocessor.preprocess(nb, {})
+
+    # attachment detection tests
+    def test_attachment_in_mark_scheme(self, preprocessor):
+        """Is an error raised when there is an attachment in the marking scheme?"""
+        source = dedent(
+            """
+            assert True
+            ### BEGIN MARK SCHEME
+            ![](attachment:image.png)
+            ### END MARK SCHEME
+            """
+        ).strip()
+        cell = create_task_cell(source, 'markdown', 'some-task-id', 2)
+        with pytest.raises(RuntimeError):
+            preprocessor._remove_mark_scheme_region(cell)
+
+        source = dedent(
+            """
+            assert True
+            ### BEGIN MARK SCHEME
+            Text text text text text.
+            ![](attachment:image1.jpg)
+            Grade grade grade.
+            ![](attachment:image2.png)
+            Mark mark mark mark.
+            ### END MARK SCHEME
```

Comment:
```suggestion
            ### BEGIN MARK SCHEME
            Text text text text text.
            ![alternative text](attachment:image1.jpg)
            Grade grade grade.
            ### END MARK SCHEME
```

Thanks @tuncbkose, I was more thinking about alternative text to the image, to test the regular expression.
But if you think it is relevant to keep also a test with 2 inserted image we can also keep it (maybe keeping text around is also a good thing)

Reply:
Thanks for the clarification. I'll add one with an alternative text. As for this example with two attachments and text around them, I don't think it matters too much but it doesn't hurt to keep it.